src/lib/address_resolve/AddressResolve_DefaultImpl.h

```diff
@@ -116,15 +147,20 @@ class NodeLookupHandle : public NodeLookupHandleBase
     /// any active searches)
     NodeLookupAction NextAction(System::Clock::Timestamp now);
 
+    /// Does the node has any valid lookup result
+    bool HasLookupResult() const { return mResults.HasValidResult(); }
+
+    /// Return the next valid lookup result.
+    ResolveResult GetLookupResult() { return mResults.ConsumeResult(); }
```

Comment:
Feels like this should be called `TakeLookupResult` to make it clear that it advances the state.